eventuate-tram-consumer-http-micronaut/src/main/java/io/eventuate/tram/consumer/http/HeartbeatService.java

```diff
@@ -0,0 +1,58 @@
+package io.eventuate.tram.consumer.http;
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker;
+import io.github.resilience4j.retry.Retry;
+import io.micronaut.context.annotation.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.PostConstruct;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Context
+public class HeartbeatService {
+  private Logger logger = LoggerFactory.getLogger(getClass());
+
+  private CircuitBreaker circuitBreaker;
+  private Retry retry;
+  private ProxyClient proxyClient;
+  private int interval;
+  private Timer timer = new Timer();
+  private ConcurrentHashMap<String, Runnable> heartbeatCalls = new ConcurrentHashMap<>();
+
+  public HeartbeatService(CircuitBreaker circuitBreaker,
+                          Retry retry,
+                          ProxyClient proxyClient,
+                          HttpConsumerProperties httpConsumerProperties) {
+    this.circuitBreaker = circuitBreaker;
+    this.retry = retry;
+    this.proxyClient = proxyClient;
+    this.interval = httpConsumerProperties.getHeartBeatInterval();
+  }
+
+  public void addSubscription(String subscriptionInstanceId) {
+    //circuitBreaker is one for all subscriptions, if we have 5 subscriptions, and circuit breaker calls is limited by 3
+    //if heartbeat fails for first 3 subscription/retry calls, other heartbeats/retries will be blocked
+    //so we can wrap each heartbeat individually
+    heartbeatCalls.put(subscriptionInstanceId,
+            Retry.decorateRunnable(retry,
```

Comment:
Please check the javadocs to see how Timer handles exceptions.
I think exceptions need to be caught, logged and ignored?